src/Utils.f90

```diff
@@ -135,6 +137,109 @@ subroutine zassert_shape(A, size, routine, matname)
       return
    end subroutine zassert_shape
 
+   subroutine print_mat(m, n, A, name)
+      !! Utility function to print a m-by-n matrix with a title
+      integer , intent(in)                         :: n
+      integer , intent(in)                         :: m
+      real (kind=wp)  , intent(in)                 :: a(m,n)
+      character(*) , optional,  intent(in)         :: name
+
+      ! internal variables
+      real (kind=wp) :: amin, amax
+      character ( len = 10 ) iform
+      integer :: i, ihi, ilo, j, jhi, jlo, lmax, npline
+      logical :: integ
+      
+      write(*,*)
+      if (present(name)) then
+         write(*,*) 'Output matrix: ', trim(name)
+      endif
+      
+      ! Check if all entries are integral. 
```

Comment:
integral? Shouldn't the comment read `integer`?

Reply:
Yeah, I guess it's synonymous. I have no preference.